app/services/claims/fetch_eligible_misc_fee_types.rb

```diff
@@ -66,15 +62,17 @@ def lgfs_trial_fee_misc_fee_types
       Fee::MiscFeeType.lgfs.where(unique_code: LGFS_TRIAL_CLAIM_ELIGIBILITY)
     end
 
-    def trial_fee_filter(relation)
-      return relation if case_type&.is_trial_fee? || case_type.nil?
-      relation.without_trial_fee_only
+    def apply_trial_fee_filter?
+      !case_type&.is_trial_fee? && case_type
```

Comment:
if you test case_type first could we get away with removing the safe navigator?
`case_type && !case_type.is_trial_fee?`